backend/API/swagger_API.yaml

```diff
@@ -0,0 +1,238 @@
+      operationId: "addStudentFeedback"
+      consumes:
+      - "application/json"
+      produces:
+      - "integer"
+      parameters:
+      - in: path
+        name: studentID
+        type: integer
+        required: true
+        description: "student ID number"
+      - in: "body"
+        name: "body"
+        description: "Student Feedback"
+        required: false
+        schema:
+          $ref: "#/definitions/Student"
+      responses:
+        "200":
+          description: "OK"
+        "400":
+          description: "Bad Request"     
+        "401":
+          description: "Unauthorized" 
+        "403":
+          description: "Forbidden"
+        "500":
+          description: "Internal Server Error"
+    get:
+      tags:
+      - "Student"
+      summary: "Get student feedback"
+      description: "Get all student feedback data"
+      operationId: "getStudentFeedback"
+      produces: 
+      - "application/json"
+      parameters: 
+      - in: path
+        name: studentID
+        type: integer
+        required: true
+        description: "student ID number"
+      responses: 
+        "200":
+          description: "OK"
+          schema:
+            type: "array"
+            items:
+              $ref: "#/definitions/Student"
+        "401":
+          description: "Unauthorized" 
+        "403":
+          description: "Forbidden"
+        "500":
+          description: "Internal Server Error"
+    patch:
+      tags:
+      - "Student"
+      summary: "Add additional feedback"
+      description: "Add additioinal feedback for existing student"
+      operationId: "updateStudent"      
+      consumes:
+      - "application/json"
+      produces:
+      - "integer"
+      parameters: 
+      - in: path
+        name: studentID
+        type: integer
+        required: true
+        description: "student ID number"
+      - in: "body"
+        name: "body"
+        description: "Student Feedback"
+        required: true
+        schema:
+          $ref: "#/definitions/Student"        
+      responses: 
+        "200":
+          description: "OK"      
+        "401":
+          description: "Unauthorized" 
+        "403":
+          description: "Forbidden"
+        "405":
+          description: "Method Not Allowed"
+    delete:
+      tags:
+      - "Student"
+      summary: "Remove Student & Feedback"
+      description: "Delete student, and all comments for student"
+      operationId: "removeStudent"      
+      parameters:
+      - in: path
+        name: studentID
+        type: integer
+        required: true
+        description: "student ID number"
+      responses:
+        "200":
+          description: "OK"        
+        "401":
+          description: "Unauthorized" 
+        "403":
+          description: "Forbidden"
+ 
+
+  /Student/{studentID}/comment/:
+    post:
+      tags:
+      - "Comment"
+      summary: "Get comment feedback for student"
```

Comment:
The summary/description describe the POST method as "get comment feedback for a particular student"--Wouldn't the POST method be for adding a comment, or is that what the Student/PATCH api is for? 

I'm unclear what the POST would be adding via /Student/{studentID}/comment/.

Reply:
Very helpful feedback. Using POST is a confusing mechanism to get a response, even though we need to provide some data. It's also unclear whether the /comment/ endpoint is for adding a new comment or getting a comment from the API.

To address these issues, I changed the request from a POST to GET, and the name of the endpoint to aifeedback.